### ISSUE TITLE
npm run network-audit hangs up due to network::switches::kLogNetLog

### ISSUE THREAD

**Issue by mariospr** [brave/brave-browser#19527]
## Description 

This newly added browser test suite is not working as expected on `master` because it's always hanging up due to the `network::switches::kLogNetLog` switch that it uses to log all net events, and the process them, causing the test to eventually fail on CI when the (really long) timeout set is reached.

Additionally, this makes difficult to add new tests to the network-audit for obvious reasons (i.e. test never ends)

## Steps to Reproduce

On a freshly built checkout of Brave 1.34.20: run `npm run network-audit` and wait ~15 min for it to finish

## Actual result:

The test fails and throws different errors on screen, most relevant ones being:
```
00:26:40  [431746:431749:1116/232618.862946:FATAL:proxy_server.cc(116)] Check failed: !is_direct(). 
00:26:40  [430101:430101:1116/232619.040418:ERROR:network_service_instance_impl.cc(972)] Network service crashed, restarting service.
00:26:40  [430101:430101:1116/232619.047112:INFO:get_parameters.cc(53)] Unexpected HTTP status: -1
00:26:40  content/public/test/test_utils.cc:399: Failure
```

## Expected result:

Test should pass after ~15min running (it does 3 sub-tests, waiting ~5 min for each of them)

## Reproduces how often: 

Always

## Brave version (brave://version info)

Brave 1.34.20 (built from sources)

**Comment by mariospr** [comment]
I already investigated this and the problem was introduced with https://github.com/brave/brave-core/commit/a251f29e0e3d5c86ee94bf54ff7e9a69b209ab1e, because our override of `ProxyServerToPacResultElement()` attempts to incorporate auth information to the host of a proxy with the custom direct:// scheme, which is not meant to work as a proxy anyway (see [upstream docs](https://chromium.googlesource.com/chromium/src/+/HEAD/net/docs/proxy.md#direct-proxy-scheme)).

Thus, we need to early return with the upstream behaviour (i.e. not trying to add any auth info) when the proxy server has a SCHEME_DIRECT scheme, probably do it also for `ProxyServerToProxyUri()`. I'll prepare a PR for that.

**Comment by mkarolin** [comment]
The fix will need to be uplifted to 1.32.x
